community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -636,12 +637,15 @@ private TransactionLogModule buildTransactionLogs(
             return -1;
         };
         final LogFileInformation logFileInformation =
-                new PhysicalLogFileInformation( logFiles, logHeaderCache, transactionIdStore::getLastCommittedTransactionId, logInformation );
+                new PhysicalLogFileInformation( logFiles, logHeaderCache,
+                        transactionIdStore::getLastCommittedTransactionId, logInformation );
 
-        String pruningConf = config.get(
-                config.get( GraphDatabaseFacadeFactory.Configuration.ephemeral )
-                ? GraphDatabaseFacadeFactory.Configuration.ephemeral_keep_logical_logs
-                : GraphDatabaseSettings.keep_logical_logs );
+        if ( config.get( GraphDatabaseFacadeFactory.Configuration.ephemeral ) )
+        {
+            config = config.withDefaults(
+                    stringMap( GraphDatabaseSettings.keep_logical_logs.name(), "1 files" ) );
```

Comment:
This seems an odd place to define this default. I don't understand why it's necessary.